numba/tests/test_np_functions.py

```diff
@@ -3172,6 +3176,66 @@ def test_1D_weights_axis(data, axis, weights):
         #test with axis argument
         test_1D_weights_axis(data, axis=1, weights=w)
 
+    def test_allclose(self):
+
+        pyfunc = np_allclose
+        cfunc = jit(nopython=True)(pyfunc)
+
+        a = np.asarray([1e10, 1e-7])
+        b = np.asarray([1.00001e10, 1e-8])
+        assert not cfunc(a, b)
+
+        a = np.asarray([1e10, 1e-8])
+        b = np.asarray([1.00001e10, 1e-9])
+        assert cfunc(a, b)
+
+        a = np.asarray([1e10, 1e-8])
+        b = np.asarray([1.0001e10, 1e-9])
+        assert not cfunc(a, b)
+
+        a = np.asarray([1e10])
+        b = np.asarray([1.0001e10, 1e-9])
+        assert not cfunc(a, b)
+
+        a = np.asarray([1e10, 1e-9, np.nan])
+        b = np.asarray([1.0001e10, 1e-9])
+        with self.assertRaises(ValueError):
+            cfunc(a, b)
```

Comment:
Move the test that raises an exception to its own function. If I'm not wrong, one need to disable the leak check when asserting exceptions:

```python

def test_allcase_exception(self):
    self.disable_leak_check()
    a = np.asarray([1e10, 1e-9, np.nan])
    b = np.asarray([1.0001e10, 1e-9])
    with self.assertRaises(ValueError) as e:
        cfunc(a, b)
    self.assertIn("put here the expected exception msg", str(e.exception))
```